File: packages/db/scripts/seed-test-user.ts
import { db } from "../db";
import { usersTable } from "../schema";
import { eq } from "drizzle-orm";
import argon2 from "argon2";

/**
 * Seed a default test user for development / demo purposes.
 *
 * Email:    mundo.dev@cv-analyzer.local
 * Password: f3l!pe_p@llm@
 * Name:     Mundo Dev (display name only)
 */

const TEST_USER = {
  email: "mundo.dev@cv-analyzer.local",
  password: "f3l!pe_p@llm@",
  isAdmin: true,
};

async function main() {
  console.log("==> Seeding test user...");

  // Check if user already exists (idempotent)
  const existing = await db
    .select()
    .from(usersTable)
    .where(eq(usersTable.email, TEST_USER.email));

  if (existing.length > 0) {
    console.log(`    Test user already exists: ${TEST_USER.email}`);
    console.log("    You can sign in with:");
    console.log(`      Email:    ${TEST_USER.email}`);
    console.log(`      Password: ${TEST_USER.password}`);
    return;
  }

  const passwordHash = await argon2.hash(TEST_USER.password);
  const result = await db
    .insert(usersTable)
    .values({
      email: TEST_USER.email,
      passwordHash,
      isAdmin: TEST_USER.isAdmin,
    })
    .returning({ id: usersTable.id });

  console.log(`    Created test user: ${result[0].id}`);
  console.log("    Sign-in credentials:");
  console.log(`      Email:    ${TEST_USER.email}`);
  console.log(`      Password: ${TEST_USER.password}`);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: apps/ui/scripts/seed_test_user.py
#!/usr/bin/env python3
"""Seed the standard local test user."""

from __future__ import annotations

import sys
from pathlib import Path

_SRC = Path(__file__).resolve().parents[1] / "src"
if str(_SRC) not in sys.path:
    sys.path.insert(0, str(_SRC))

from ui.auth.users import create_user, get_user_by_email

EMAIL = "mundo.dev@cv-analyzer.local"
PASSWORD = "f3l!pe_p@llm@"


def main() -> int:
    existing = get_user_by_email(EMAIL)
    if existing is not None:
        print(f"Test user already exists: {existing.id}")
        return 0
    try:
        user_id = create_user(EMAIL, PASSWORD, is_admin=True)
    except Exception as exc:  # noqa: BLE001
        print(f"Failed to seed test user: {exc}", file=sys.stderr)
        return 1
    print(f"Created test user {user_id}")
    print(f"  email: {EMAIL}")
    print(f"  password: {PASSWORD}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
